frontend/server/controllers/CourseController.php

```diff
@@ -302,17 +326,17 @@ public static function apiCreate(Request $r) {
             ]));
 
             // Create the actual course
-            CoursesDAO::save(new Courses([
+            CoursesDAO::create(new Courses([
                 'name' => $r['name'],
                 'description' => $r['description'],
                 'alias' => $r['alias'],
                 'group_id' => $group->group_id,
                 'acl_id' => $acl->acl_id,
-                'school_id' => is_null($r['school']) ? null : $r['school']->school_id,
+                'school_id' => $r['school_id'] ?? null,
                 'start_time' => gmdate('Y-m-d H:i:s', $r['start_time']),
                 'finish_time' => gmdate('Y-m-d H:i:s', $r['finish_time']),
-                'public' => is_null($r['public']) ? false : $r['public'],
-                'show_scoreboard' =>$r['show_scoreboard'] == 'true',
+                'public' => $r['public'],
```

Comment:
```suggestion
                'public' => boolval($r['public']),
```

eso se asegura que no sea nulo.

Reply:
listo